test/Reserves.t.sol

```diff
@@ -45,6 +40,6 @@ contract ReservesTest is Test {
     function test_fuzz_get_set(Currency currency, uint256 value) public {
         vm.assume(value != type(uint256).max);
         currency.setReserves(value);
-        assertEq(currency.getReserves(), value);
+        assertEq(Reserves.getReserves(), value);
```

Comment:
these tests should add checking the getter for the currency!
Yu have `currency.setReserves(value)` so then we need to check
- is value set correctly (you have this)
- is currency set correctly (this needs to be added)

Reply:
Added checks for the currency in the tests and added a new test to confirm the functionality of `Reserves.reset()`